src/pages/RecipeIndex.js

```diff
@@ -1,43 +1,134 @@
-import React from 'react'
-import { NavLink } from 'react-router-dom';
+import {React, useState}from 'react';
+import SearchBar from '../components/SearchBar';
+import { NavLink, Link } from 'react-router-dom';
 import { Card, CardBody, CardSubtitle, CardText, CardTitle, Button } from 'reactstrap';
 
-
 const RecipeIndex = ({ recipes }) => {
+  const [filteredRecipes, setFilteredRecipes] = useState(recipes)
+
+  const handleSearchChange = (filteredResults) => {
+    setFilteredRecipes(filteredResults)
+  }
+  
   return (
     <>
     <h1>Disaster Recipes</h1>
+    <SearchBar recipes={recipes} onSearchChange={handleSearchChange}/>
+
     <div>
-      {recipes?.map((recipe, index) => {
-      return (
-        <Card key={index} style={{width: '16rem'}} outline color='dark'>
-          <img alt={recipe.name} src={recipe.recipe_image}/>
-          <CardBody>
-            <CardTitle tag="h5">
-              {recipe.name}
-            </CardTitle>
-            <CardSubtitle
-              className="mb-2 text-muted"
-              tag="h6"
-            >
-            Courses: {recipe.course}
-            </CardSubtitle>
-            <CardText>
-              Cook Time: {recipe.cook_time}, 
-              Servings: {recipe.servings}
-            </CardText>
-            <NavLink to={`/recipe/${recipe.id}`}>
-              <Button aria-label='Disaster Info'>
-                  Disaster Info
-              </Button>
-            </NavLink>
-          </CardBody>
-        </Card>
-      )
-      })}
+     <div className="indexHeading">
+        <h2>Desserts</h2>
+        <p>HMMMMM DIABETES</p>
+      </div>
+      <div className="indexCards">
+        {filteredRecipes.map((recipe, index) => {
+          if (recipe.course === 'Dessert') {
+            return (
+              <Link to={"/contactus"}>
+              <Card key={index} style={{ width: '16rem' }} outline color='dark'>
+                <img alt={recipe.name} src={recipe.recipe_image} />
+                <CardBody>
+                  <CardTitle tag="h5">
+                    {recipe.name}
+                  </CardTitle>
+                  <CardSubtitle
+                    className="mb-2 text-muted"
+                    tag="h6"
+                  >
+                    Courses: {recipe.course}
+                  </CardSubtitle>
+                  <CardText>
+                    Cook Time: {recipe.cook_time}, 
+                    Servings: {recipe.servings}
+                  </CardText>
+                  <NavLink to={`/recipe/${recipe.id}`}>
+                    <Button aria-label='Disaster Info'>
+                      Disaster Info
+                    </Button>
+                  </NavLink>
+                </CardBody>
+              </Card>
+              </Link>
+            )
+          }
+          return null
```

Comment:
Consider returning something that is more informative to the user and lets them know there is no recipe with that search criteria.